server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -1077,9 +1084,11 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeByte(state.id());
         writeSettingsToStream(settings, out);
         out.writeVLongArray(primaryTerms);
-        out.writeVInt(mappings.size());
-        for (MappingMetadata mappingMetadata : mappings.values()) {
-            mappingMetadata.writeTo(out);
+        if (mapping == null) {
```

Comment:
Maybe add a TODO here and in `readFrom(...)` method that we should eventually change the serialization format.